plugins/Referrers/Columns/Base.php

```diff
@@ -137,6 +139,32 @@ protected function getReferrerInformation($referrerUrl, $currentUrl, $idSite, Re
         return $referrerInformation;
     }
 
+    protected function excludeQueryParamsFromReferrerUrl()
+    {
+        // Add additional excludes for specific referrers
+        Piwik::addAction('Tracker.PageUrl.getQueryParametersToExclude', function(&$parametersToExclude) {
+            if (!empty($this->referrerHost) && strpos($this->referrerHost, 'instagram.com') !== false) {
+                $parametersToExclude[] = 'e';
+                $parametersToExclude[] = 's';
+            }
+            if (!empty($this->referrerHost) && strpos($this->referrerHost, 'facebook.com') !== false) {
+                $parametersToExclude[] = 'h';
+            }
+            if (!empty($this->referrerHost) && strpos($this->referrerHost, 'google.') !== false) {
+                $parametersToExclude[] = 'ust';
```

Comment:
fyi some tests seem to be failing and need to be updated.

In https://travis-ci.org/github/matomo-org/matomo/jobs/683276069#L984

wonder if we can also exclude `bvm` and `ei`.

Re the addAction... wondering if this causes issues eg in BulkTracking. Say it calls

* PageUrl::excludeQueryParametersFromUrl($pageUrl, $this->idsite);
* AddAction to exclude more params
* PageUrl::excludeQueryParametersFromUrl($this->referrerUrl, $this->idsite);
* now it does another tracking request within bulk tracking
* PageUrl::excludeQueryParametersFromUrl($pageUrl, $this->idsite); // now it would potentially also exclude above parameters if referrers was one of these?

Reply:
Could maybe add a third parameter say `excludeQueryParametersFromUrl($originalUrl, $idSite, $extraParamsToExclude = array())`?

Reply:
Changed the implementation to use an additional parameter as suggested.